heroku/heroku-builds
# Updated all 3rd party deps and updated project to TS

This PR updates 3rd party dep, updates the project to TypeScript and fixes broken unit tests.

### Build/Install Instructions
1. clone this repo and checkout this branch
2. using node 18+ run `yarn && yarn build`
3. cd into `dist/` and run `npm pack`
4. Copy the *absolute path* to the tar created in `dist/`
5. install it using `heroku plugins:add "file://path-to/heroku-builds/dist/heroku-cli-heroku-builds-0.0.30.tgz"`
6. verify commands are available by running `heroku commands | grep build:`

### Test Instructions
1. `heroku builds -a YOUR_APP` // Lists 15 most recently created builds (API Returns all builds)
2. `heroku builds -n 50 -a YOUR_APP` // Optionally show more builds with the `-n` flag (API Returns all builds)
3. `heroku builds:info -a YOUR_APP` // Show details of the latest build
4. `heroku builds:info BUILD_ID -a YOUR_APP` // Show details of a specific build
5. `heroku builds:create -a YOUR_APP` // Create a .tar.gz from the contents of the cwd, upload and create a build
6.  `heroku builds:create -a YOUR_APP --source-tar PATH_TO_TARBALL` // Specify a tarball to build from
7.  `heroku builds:output -a YOUR_APP` // Show last build output 
8. `heroku builds:output BUILD_ID -a YOUR_APP` // Show build output for specified build id
9. `heroku builds:cancel BUILD_ID -a YOUR_APP` // cancel a build

Please note that 1 and 2 in the list seems to always return *all* builds. This behavior is confirmed with the older plugin as well.
